src/rejson_api.h

```diff
@@ -85,6 +85,10 @@ typedef struct RedisJSONAPI {
   int (*pathIsSingle)(JSONPath);
   int (*pathHasDefinedOrder)(JSONPath);
 
+  // Return JSON String representation from an iterator (without consuming the iterator)
+  // The caller gains ownership of `str`
+  int (*getJSONFromIter)(JSONResultsIterator iter, RedisModuleCtx *ctx, RedisModuleString **str);
```

Comment:
IMO, this is not the right solution since once we run aggregate function, we need to change the values which this solution does not allow. 
Maybe we can provide a replyWithJson API call.

Reply:
This PR is about projection, and the API `getJSONFromIter` is satisfying the requirement to serialize read-only JSON data, which is owned by the JSON API.
In addition, `getJSONFromIter` allowing to get a JSON string representation of an iterator, IMO seems useful and worthy independently of this PR, and it nicely complements `getJSON` API.

I agree that for aggregation, different APIs would be required in order to serialize read-write data (owned and manipulated by the aggregation pipeline, but I don't think it should be done in this PR.